aat/benchmark_rpc_test.go

```diff
@@ -148,27 +148,38 @@ func benchmarkRpc(b *testing.B, action client.InvocationHandler, callArgs wamp.L
 		panic("Failed to register procedure: " + err.Error())
 	}
 
-	client, err := connectClient()
-	if err != nil {
-		panic("Failed to connect client: " + err.Error())
-	}
-
-	ctx := context.Background()
-	var result *wamp.Result
-
 	b.ResetTimer()
 
+	clientsWg := &sync.WaitGroup{}
+
 	for i := 0; i < b.N; i++ {
-		result, err = client.Call(ctx, "action", nil, callArgs, nil, "")
-		if err != nil {
-			panic(err)
-		}
+		clientsWg.Add(1)
+		go func() {
+			client, err := connectClient()
+			defer client.Close()
+			defer clientsWg.Done()
+
+			if err != nil {
+				panic("Failed to connect client: " + err.Error())
+			}
+
+			ctx := context.Background()
+			var result *wamp.Result
+
+			for j:= 0; j < 10; j++ {
+				result, err = client.Call(ctx, "action", nil, callArgs, nil, "")
+				if err != nil {
+					panic(err)
+				}
+				verify(result)
+			}
+		}()
```

Comment:
When looking at this... it creates and connects N clients, each doing 10 RPC calls...  I was not sure if it was trying to test how fast clients could be connected, or testing how fast RPC calls were being made.  May I suggest making a new benchmark like this, except it only connects N clients - no RCP. - Probably makes sense just to do that with HTTP/HTTPS.

If I am missing an important reason for the RPC calls, please let me know.  Otherwise, it just looks like it conflates what you are really trying to benchmark.